### PR TITLE
add xcit

### PR DESCRIPTION
**Pull request by bnucsy**
Thank you for your contribution to the MindCV repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

**Comment by Songyuanwei** [comment]
目前不需要中文readme，需要删掉中文readme

File: benchmark_results.md [hunk 1]
```diff
@@ -74,6 +74,7 @@
 | vit_l_16_224 | D910x8-G | 76.34     | 92.79     | 303.31     | [yaml](https://github.com/mindspore-lab/mindcv/blob/main/configs/vit/vit_l16_224_ascend.yaml) | [weights](https://download.mindspore.cn/toolkits/mindcv/vit/vit_l_16_224.ckpt) |
 | vit_l_32_224 | D910x8-G | 73.71     | 90.92     | 305.52     | [yaml](https://github.com/mindspore-lab/mindcv/blob/main/configs/vit/vit_b32_224_ascend.yaml) | [weights](https://download.mindspore.cn/toolkits/mindcv/vit/vit_l_32_224.ckpt) |
 | Xception | D910x8-G | 79.01     | 94.25     | 22.91      | [yaml](https://github.com/mindspore-lab/mindcv/blob/main/configs/xception/xception_ascend.yaml) | [weights](https://download.mindspore.cn/toolkits/mindcv/xception/xception_299.ckpt) |
+| xcit_tiny_12_p16 | D910x8-G | 77.674     | 93.792      | 7      | [yaml](https://github.com/bnucsy/mindcv/tree/main/configs/xcit/xcit_tiny_12_p16_ascend.yaml) | [weights](https://download.mindspore.cn/toolkits/mindcv/xcit/xcit_tiny_12_p16_best.ckpt) |
```
Comment (Songyuanwei) [inline benchmark_results.md:77]: 目前不需要修改提交benchmark_results.md, 后续会统一运行提交

File: configs/xcit/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+## Results
+
+Our reproduced model performance on ImageNet-1K is reported as follows.
+
+<div align="center">
+
+| Model        | Context  | Top-1 (%) | Top-5 (%) | Params (M) | Recipe                                                                                        | Download                                                                       |
+|--------------|----------|-----------|-----------|------------|-----------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------|
+| xcit_tiny_12_p16 | D910x8-G | 77.674     | 93.792      | 7      | [yaml](https://github.com/bnucsy/mindcv/tree/main/configs/xcit/xcit_tiny_12_p16_ascend.yaml) | [weights](https://download.mindspore.cn/toolkits/mindcv/xcit/xcit_tiny_12_p16_best.ckpt) |
```
Comment (Songyuanwei) [inline configs/xcit/README.md:34]: Top-1, Top-5, params 保留至小数点后两位，例如77.67， 93.79

Comment (Songyuanwei) [inline configs/xcit/README.md:34]: yaml文件的地址不要写自己仓的地址，要写主仓yaml对应的地址

File: configs/xcit/README.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+Please download the [ImageNet-1K](https://www.image-net.org/challenges/LSVRC/2012/index.php) dataset for model training and validation.
+
+### Training
+
+* Distributed Training
+
+It is easy to reproduce the reported results with the pre-defined training recipe. For distributed training on multiple Ascend 910 devices, please run
+
+```shell
+# distrubted training on multiple GPU/Ascend devices
```
Comment (Songyuanwei) [inline configs/xcit/README.md:63]: distrubted -> distributed

File: configs/xcit/xcit_tiny_12_p16_ascend.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# system config
```
Comment (Songyuanwei) [inline configs/xcit/xcit_tiny_12_p16_ascend.yaml:1]: system config -> system， 去掉config， 下同

File: configs/xcit/xcit_tiny_12_p16_ascend.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+# dataset config
+dataset: 'imagenet'
+data_dir: '/cache/dataset/imagenet/imagenet/'
+shuffle: True
+dataset_download: False
+batch_size: 128
+drop_remainder: True
+
+# Augmentation config
```
Comment (Songyuanwei) [inline configs/xcit/xcit_tiny_12_p16_ascend.yaml:15]: Augmentation config -> augmentation   首字母小写， 去掉config，下同

File: configs/xcit/xcit_tiny_12_p16_ascend.yaml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+image_resize: 224
+hflip: 0.5
+color_jitter: 0.4
+interpolation: 'bicubic'
+crop_pct: 0.875
+re_prob: 0.25
+mixup: 0.8
+cutmix: 1.0
+auto_augment: 'randaug-m9-mstd0.5-inc1'
+use_ema: True
```
Comment (Songyuanwei) [inline configs/xcit/xcit_tiny_12_p16_ascend.yaml:25]: use_ema -> ema,   参数名已经从use_ema更新为ema

File: configs/xcit/xcit_tiny_12_p16_ascend.yaml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+ema_decay: 0.99996
+
+# model config
+model: 'xcit_tiny_12_p16'
+num_classes: 1000
+pretrained: False
+ckpt_path: ""
+keep_checkpoint_max: 30
+save_checkpoint: True
+ckpt_save_dir: './xcit_tiny_12_p16'
```
Comment (Songyuanwei) [inline configs/xcit/xcit_tiny_12_p16_ascend.yaml:35]: './xcit_tiny_12_p16' -> './ckpt'

File: configs/xcit/xcit_tiny_12_p16_ascend.yaml [hunk 8]
```diff
@@ -0,0 +1,59 @@
+# system config
+mode: 0
+distribute: True
+num_parallel_workers: 16
+val_while_train: True
+
+# dataset config
+dataset: 'imagenet'
+data_dir: '/cache/dataset/imagenet/imagenet/'
```
Comment (Songyuanwei) [inline configs/xcit/xcit_tiny_12_p16_ascend.yaml:9]: '/cache/dataset/imagenet/imagenet/' -> '/path/to/imagenet'

File: configs/xcit/xcit_tiny_12_p16_ascend.yaml [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+amp_level: 'O2'
+drop_rate: 0.0
+drop_path_rate: 0.0
+
+# loss config
+loss: 'CE'
+label_smoothing: 0.1
+
+# lr scheduler config
+scheduler: 'warmup_cosine_decay'
```
Comment (Songyuanwei) [inline configs/xcit/xcit_tiny_12_p16_ascend.yaml:47]: 参数变更 'warmup_cosine_decay' -> 'cosine_decay'

File: mindcv/models/xcit.py [hunk 10]
```diff
@@ -0,0 +1,486 @@
+"""Define Cross-Covariance Image Transformer model"""
```
Comment (Songyuanwei) [inline mindcv/models/xcit.py:1]: 参考其他model的前两行, 写一下 Refer to

File: mindcv/models/xcit.py [hunk 11]
```diff
@@ -0,0 +1,486 @@
...
+        'url': url,
+        'num_classes': 1000,
+        'first_conv': '',
+        'classifier': '',
+        **kwargs
+    }
+
+
+default_cfgs = {
+    'xcit_tiny_12_p16': _cfg(url=''),
```
Comment (Songyuanwei) [inline mindcv/models/xcit.py:34]: ckpt的链接

File: configs/xcit/README.md [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+  <em>Figure 1. Architecture of XCiT [<a href="#references">1</a>] </em>
+</p>
+
+
+* **Linear complexity in time and memory**
+
+  XCiT models has a linear complexity w.r.t number of patches/tokens: $\mathcal{O}(N d ^2)$
+
+
+![](https://user-images.githubusercontent.com/51260139/211969388-0658c89b-c41c-4df9-a295-5b3431b626b7.png)  |  ![](https://user-images.githubusercontent.com/51260139/211969950-92b15d1d-0b08-4075-9a12-faf40cd49efa.png)
```
Comment (Songyuanwei) [inline configs/xcit/README.md:21]: Linear complexity in time and memory  及本张图片等相关内容删掉，与mindspore无关

**Review by geniuspatrick — APPROVED** [review]
